src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -4335,6 +4335,15 @@ void MacroAssembler::shadd(Register Rd, Register Rs1, Register Rs2, Register tmp
   }
 }
 
+void MacroAssembler::slli_uw(Register Rd, Register Rs, int shamt) {
+  if (UseZba) {
+    _slli_uw(Rd, Rs, shamt);
+  } else {
+    slli(Rd, Rs, shamt + 32);
+    srli(Rd, Rd, 32);
```

Comment:
I don't think this code in else block will work when `shamt` >= 32. Note that the `slli.uw` instruction is the same as `slli` with `zext.w` performed on the least-significant word of `Rs` before shifting. So you might want to do a combination of `zext.w` and `slli` on `Rs` instead.

Reply:
slli.uw with shamt >= 32 will be same as ```add Rd, X0, X0```, isn't it ?
He can just add handling for that special case then